Godeps/_workspace/src/github.com/openshift/source-to-image/pkg/api/types.go

```diff
@@ -91,6 +96,11 @@ type Config struct {
 	// Specify a relative directory inside the application repository that should
 	// be used as a root directory for the application.
 	ContextDir string
+
+	// AllowedUIDs is a list of user ranges of users allowed to run the builder image.
+	// If a range is specified and the builder image uses a non-numeric user or a user
+	// that is outside the specified range, then the build fails.
+	AllowedUIDs user.RangeList
```

Comment:
Basically yes

On Fri, Aug 7, 2015 at 5:03 PM, Paul Weil <notifications@github.com> wrote:

> In
> Godeps/_workspace/src/github.com/openshift/source-to-image/pkg/api/types.go
> <https://github.com/openshift/origin/pull/4025#discussion_r36561052>:
>
> > @@ -91,6 +96,11 @@ type Config struct {
> >  	// Specify a relative directory inside the application repository that should
> >  	// be used as a root directory for the application.
> >  	ContextDir string
> > +
> > +	// AllowedUIDs is a list of user ranges of users allowed to run the builder image.
> > +	// If a range is specified and the builder image uses a non-numeric user or a user
> > +	// that is outside the specified range, then the build fails.
> > +	AllowedUIDs user.RangeList
>
> Yes, I believe this will be sufficient. If I'm understanding the sequence
> from the code below it is basically
>
>    1. See if the service account has access to run build pods as root by
>    trying to admit a pod with runAsUser forced to 0
>    2. If the above fails set a range of anything except 0 with 1-
>    3. The STI builder checks the build image to prevent it from running
>    if the uid is not in the range set by the env var.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4025/files#r36561052>.
>